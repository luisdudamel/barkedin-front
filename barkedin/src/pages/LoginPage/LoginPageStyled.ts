import styled from "styled-components";

const LoginPageStyled = styled.div`
  button {
    background-color: #2a9d8f;
    width: 100px;
    border-radius: 20px;
  }
  .avatars-container {
    display: flex;
  }
  h1 {
    padding-top: 30px;
  }

  @media (max-width: 440px) {
    .login-title {
      font-size: 20px;
    }
    .MuiAvatar-img {
      width: 50px;
      height: 50px;
    }
    .MuiAvatar-root {
      width: 50px;
      height: 50px;
    }
  }
  @media (min-width: 441px) {
    .login-input {
      width: 300px;
    }
    .form-container {
      justify-content: center;
    }
    .form-link {
      font-size: 20px;
    }
  }
`;
export default LoginPageStyled;
