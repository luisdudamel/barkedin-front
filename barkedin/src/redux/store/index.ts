import { configureStore } from "@reduxjs/toolkit";
import dogsSliceReducer from "../feature/dogsSlice";
import uiSliceReducer from "../feature/uiSlice";
import userSliceReducer from "../feature/usersSlice";

const store = configureStore({
  reducer: {
    user: userSliceReducer,
    ui: uiSliceReducer,
    dogs: dogsSliceReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
