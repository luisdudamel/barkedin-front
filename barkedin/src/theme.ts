import { createTheme } from "@mui/material/styles";
import { red } from "@mui/material/colors";

const theme = createTheme({
  typography: {
    fontFamily: ["Outfit"].join(","),
  },
  components: {
    MuiButton: {
      styleOverrides: {
        root: {
          textTransform: "none",
          "&:hover": {
            backgroundColor: "white",
            color: "#F4A261",
          },
        },
      },
    },
    MuiAppBar: {
      styleOverrides: {
        colorPrimary: {
          backgroundColor: "white",
          color: "#264653",
        },
      },
    },
  },
  palette: {
    background: {
      default: "#ffc027",
    },
    primary: {
      main: "#264653",
    },
    secondary: {
      main: "#19857b",
    },
    error: {
      main: red.A400,
    },
  },
});

export default theme;
