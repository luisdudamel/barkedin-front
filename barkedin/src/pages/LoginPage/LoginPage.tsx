import LoginForm from "../../components/LoginForm/LoginForm";
import LoginPageStyled from "./LoginPageStyled";

const LoginPage = (): JSX.Element => {
  return (
    <LoginPageStyled>
      <LoginForm />
    </LoginPageStyled>
  );
};

export default LoginPage;
