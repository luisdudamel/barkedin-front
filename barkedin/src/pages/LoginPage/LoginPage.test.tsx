import { ThemeProvider } from "@mui/material";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import TestRenderer from "react-test-renderer";
import store from "../../redux/store";
import theme from "../../theme";
import LoginPage from "./LoginPage";

describe("Given a LoginPage component page", () => {
  describe("When it's invoked", () => {
    test("Then it should always match this snapshot", () => {
      const testedRegisterPage = TestRenderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <ThemeProvider theme={theme}>
              <LoginPage />
            </ThemeProvider>
          </BrowserRouter>
        </Provider>
      );

      expect(testedRegisterPage).toMatchSnapshot();
    });

    test("Then it should render a register form with a button with the text 'Login'", () => {
      render(
        <Provider store={store}>
          <BrowserRouter>
            <ThemeProvider theme={theme}>
              <LoginPage />
            </ThemeProvider>
          </BrowserRouter>
        </Provider>
      );

      const expectedButton = screen.getByRole("button", { name: "Login" });

      expect(expectedButton).toBeInTheDocument();
    });
  });
});
