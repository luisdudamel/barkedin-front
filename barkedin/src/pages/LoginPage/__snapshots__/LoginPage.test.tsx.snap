// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a LoginPage component page When it's invoked Then it should always match this snapshot 1`] = `
<div
  className="sc-bczRLJ VRLlR"
>
  <main
    className="MuiContainer-root MuiContainer-maxWidthLg form-container css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiBox-root css-1lhe3be"
    >
      <div
        className="avatars-container"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular css-3cgv3-MuiAvatar-root"
        >
          <img
            alt="Doberman avatar icon"
            className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="../../images/doberman.webp"
          />
        </div>
        <div
          className="MuiAvatar-root MuiAvatar-circular css-3cgv3-MuiAvatar-root"
        >
          <img
            alt="Husky avatar icon"
            className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="../../images/dog.webp"
          />
        </div>
        <div
          className="MuiAvatar-root MuiAvatar-circular css-3cgv3-MuiAvatar-root"
        >
          <img
            alt="Greyhound avatar icon"
            className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="../../images/greyhound.webp"
          />
        </div>
      </div>
      <h1
        className="MuiTypography-root MuiTypography-h5 login-title css-1icvlpi-MuiTypography-root"
      >
        Welcome to BarkedIn!
      </h1>
      <form
        className="MuiBox-root css-1y9cz7a"
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root login-input css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-15ycsir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="username"
            id="username-label"
          >
            Username
            <span
              aria-hidden={true}
              className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-dqj5rd-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-hiddenLabel css-1hybmvl-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputHiddenLabel css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="username"
              name="username"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Username"
              required={true}
              type="username"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Username
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root login-input css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-15ycsir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="password"
            id="password-label"
          >
            Password
            <span
              aria-hidden={true}
              className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-dqj5rd-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-hiddenLabel css-1hybmvl-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="current-password"
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputHiddenLabel css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="password"
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="password"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Password
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          className="MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-fullWidth MuiButtonBase-root Mui-disabled css-u6udi5-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          disabled={true}
          id=":r2:"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="submit"
        >
          Login
        </button>
        <div
          className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          style={
            Object {
              "justifyContent": "center",
            }
          }
        >
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <a
              className="form-link"
              href="/register"
              onClick={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              Not a member? Register here
            </a>
          </div>
        </div>
      </form>
    </div>
  </main>
</div>
`;
